src/L3l0Labs/SystemAccess/UseCase/RegisterUser.php

```diff
@@ -0,0 +1,43 @@
+<?php declare (strict_types = 1);
+ 
+namespace L3l0Labs\SystemAccess\UseCase;
+
+use L3l0Labs\Accounting\Invoice\VatIdNumber;
+use L3l0Labs\SystemAccess\Email;
+use L3l0Labs\SystemAccess\Exception\UserNotFoundException;
+use L3l0Labs\SystemAccess\PasswordHash;
+use L3l0Labs\SystemAccess\UseCase\RegisterUser\Command;
+use L3l0Labs\SystemAccess\User;
+use L3l0Labs\SystemAccess\UserId;
+use L3l0Labs\SystemAccess\Users;
+
+final class RegisterUser
+{
+    /**
+     * @var Users
+     */
+    private $users;
+
+    public function __construct(Users $users)
+    {
+        $this->users = $users;
+    }
+
+    public function execute(Command $command)
+    {
+        try {
+            $this->users->getByEmail(new Email($command->email()));
```

Comment:
Maybe we should add to `Users` collection interface method like `has(Email $email)`. In this method we do not have even hydrate whole user object, so it can be better for performance and memory usage reasons (of course in this app scale it should not be the problem, but it should look better as well). I would even rename `getByEmail(Email $email)` into `get(Email $email)`;